deployment-detector/src/agent.spec.ts

```diff
@@ -24,54 +30,130 @@ const createMockArg = (agentId:number, owner:string, metaData:string, chainId:nu
-    const findings:Finding[] = await handleTransaction(txEvent)
-    expect(findings).toStrictEqual([])
-  })
+  it("ignores transactions when correct deployer address calls a non-Forta contract that emits the same event", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(mockDeployerAddress)
+      .setTo(mockOtherFortaContract)
+      .addEventLog(FORTA_AGENT_EVENT, mockOtherFortaContract, [
+        mockarg.agentId,
+        mockarg.owner,
+        mockarg.metaData,
+        mockarg.chainIds,
+      ]);
+    const findings: Finding[] = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
 
-  // Test-Case-Three
+  it("creates an alert when a bot is deployed or updated from the correct deployer address and the correct contract", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(mockDeployerAddress)
+      .setTo(mockFortaAgentRegistry)
+      .addEventLog(FORTA_AGENT_EVENT, FORTA_AGENT_REGISTRY, [
+        mockarg.agentId,
+        mockarg.owner,
+        mockarg.metaData,
+        mockarg.chainIds,
+      ]);
+    const findings = await handleTransaction(txEvent);
 
-  it("returns alerts",async () => {
-    const txEvent:TransactionEvent = new TestTransactionEvent()
-    .setFrom(NETHERMIND_DEPLOYER)
-    .setTo(botsParams.proxyAddress)
-    .addTraces({
-      function: FORTA_CREATE_AGENT,
-      to:botsParams.proxyAddress,
-      arguments:[arg.agentId, arg.owner, arg.metaData, arg.chainIds]
-    })
-    const findings:Finding[] = await handleTransaction(txEvent)
-    const mockFinding = [createFinding(arg.agentId,arg.metaData,arg.owner,arg.chainIds)]
-    expect(findings).toStrictEqual(mockFinding)
-  })
+    expect(findings).toStrictEqual([
+      Finding.fromObject({
+        name: "Nethermind Forta-Bot-Deployer Detector",
+        description: `monitors bot deployments from the Nethermind deployer address`,
+        alertId: "NETH-BOT-DEPLOYER",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        protocol: "Forta",
+        metadata: {
+          agentId: mockarg.agentId.toString(),
+          by: mockarg.owner,
+          chainIds: mockarg.chainIds.toString(),
+        },
+      }),
+    ]);
+  });
 
-})
\ No newline at end of file
+  it("creates an alert  when multiple bots is deployed or updated from the correct deployer address and the correct contract", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(mockDeployerAddress)
+      .setTo(mockFortaAgentRegistry)
+      .addEventLog(FORTA_AGENT_EVENT, FORTA_AGENT_REGISTRY, [
+        mockarg.agentId,
+        mockarg.owner,
+        mockarg.metaData,
+        mockarg.chainIds,
+      ]);
+    const txEvent2: TransactionEvent = new TestTransactionEvent()
+      .setFrom(mockDeployerAddress)
+      .setTo(mockFortaAgentRegistry)
+      .addEventLog(FORTA_AGENT_EVENT, FORTA_AGENT_REGISTRY, [
```

Comment:
You could also "chain" this `.addEventLog()` method to `txEvent`, to test the bot with one transaction event rather than two.